File: python/examples/openai_tool.py
from pipe_storage import PipeStorage, openai_pipe_tools, run_openai_pipe_tool

pipe = PipeStorage()
tools = openai_pipe_tools()


def run_tool(name: str, arguments_json: str) -> dict:
    return run_openai_pipe_tool(pipe, name, arguments_json)

File: typescript/examples/openai-tool.ts
import {
  PipeStorageClient,
  createOpenAIPipeTools,
  runOpenAIPipeTool,
} from "../src/index.js";

const pipe = new PipeStorageClient();
export const tools = createOpenAIPipeTools();

export async function runTool(name: string, args: string) {
  return runOpenAIPipeTool(pipe, name, args);
}
